web/themes/custom/contribtracker/cypress.config.js

```diff
@@ -1,8 +1,56 @@
 import { defineConfig } from 'cypress';
+import { readFileSync } from 'fs';
+import neatCSV from 'neat-csv';
+import path from 'path';
+import cypressGrep from 'cypress-grep/src/plugin';
+// import cypressSplit from "cypress-split"; // Uncomment if you are using cypress-split
 
 export default defineConfig({
+  projectId: 'contrib-tracker',
+  viewportWidth: 1512,
+  viewportHeight: 1200,
+  watchForFilesChanges: false,
+  chromeWebSecurity: false,
+  defaultCommandTimeout: 60000,
+  execTimeout: 10000,
+  pageLoadTimeout: 500000,
+  requestTimeout: 90000,
+  responseTimeout: 90000,
+  experimentalStudio: true,
+  trashAssetsBeforeRuns: false,
+  experimentalShadowDomSupport: true,
+  reporter: 'mochawesome',
+  reporterOptions: {
+    useInlineDiffs: true,
+    embeddedScreenshots: true,
+    reportDir: 'cypress/results',
+    reportFilename: '[name].html',
+    overwrite: true,
+    html: true,
+    json: true,
+    charts: true,
+  },
   e2e: {
-    baseUrl: 'https://contribtracker.ddev.site/',
+    async setupNodeEvents(on, config) {
+      const inputFileName = config.env.inputTestFile;
+      console.log('loading file', inputFileName);
+      const text = readFileSync(inputFileName, 'utf8');
+      const csv = await neatCSV(text);
+      console.log('loaded the urls');
+
+      cypressGrep(on, config);
+      const version = config.env.version || 'stage';
+      // Load env from json using dynamic import
+      const envConfigPath = path.resolve(`./cypress/config/${version}.json`);
+      const envConfig = JSON.parse(readFileSync(envConfigPath, 'utf-8'));
+      config.env = { ...config.env, ...envConfig };
+      // Change baseUrl
+      config.baseUrl = config.env.baseUrl;
+      console.log(config.baseUrl);
+      config.env.urlsList = csv;
+      // cypressSplit(on, config); // Uncomment if you are using cypress-split
+      return config;
+    },
+    specPattern: 'cypress/e2e/**/*.{js,jsx,ts,tsx}',
   },
-  video: true,
```

Comment:
We don't want videos? Should we make this configurable? Anyway, it seems out of the scope for this PR.

Reply:
@hussainweb This would help a non-technical user to look through why the tests are failing. We'll see the video attached to the HTML report itself. Let me know your thoughts here.

Reply:
@KosalaiV, it was a question. You are removing the line `video: true` and I presumed that this change will disable videos from being generated.

Reply:
@KosalaiV,
> Let's keep video: true for now,

So you're going to revert this change?

Reply:
@hussainweb Will be updating it with env variable like this:
`video: process.env.CYPRESS_VIDEO !== 'false'`

The plan is to not include videos for VR scripts, hence it defaults to true for the functional test cases